core/citrus-base/src/main/java/org/citrusframework/report/LoggingReporter.java

```diff
@@ -282,6 +329,18 @@ protected void debug(String line) {
      * @return
      */
     protected boolean isDebugEnabled() {
-        return log.isDebugEnabled();
+        return enabled && log.isDebugEnabled();
+    }
+
+    /**
+     * Sets the enabled property.
+     * @param enabled
+     */
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
```

Comment:
is there a possibility to completly supress the underlying logger? because you'd have to add an immediate `return` to **every** method - also `info` and `error` etc.

Reply:
or we do say that in the case of a direct invocation it might be wanted and we do knowingly not suppress it. but that clashes with the current implementation of the `debug` method.

Reply:
maybe this https://www.slf4j.org/api/org/slf4j/helpers/NOPLogger.html can help?

Reply:
Info and Debug should also be ignore. It m

> is there a possibility to completly supress the underlying logger? because you'd have to add an immediate `return` to **every** method - also `info` and `error` etc. that seems a bit much to me....

Please see the javadoc of the class.

Reply:
> maybe this https://www.slf4j.org/api/org/slf4j/helpers/NOPLogger.html can help?

Gratefully accepted.